lua/dap.lua

```diff
@@ -451,12 +451,16 @@ function Session:event_stopped(stopped)
       self.current_frame = nil
       threads[stopped.threadId].frames = frames
       for _, frame in pairs(frames_resp.stackFrames) do
-        if not current_frame then
+        if not current_frame and frame.source and frame.source.path then
           current_frame = frame
           self.current_frame = frame
         end
         table.insert(frames, frame)
       end
+      if not current_frame then
+        current_frame = frames[1]
```

Comment:
Not sure if there will always be at least 1 frame. Theoretically there should, but maybe we should handle the empty case anways?

Reply:
This is only a fallback to previous behavior. If there is no frame, then `current_frame` is just `nil` triggering the check a few lines down.

But you are right that this behavior should be only triggered on exceptions.

Reply:
I meant that `frames` could be empty. I pushed a fixup to handle the case. If it's okay for you I'll squash it and merge